Use markdown for images in package descriptions

Edru2 opened the pull request:
#### Brief overview of PR changes/additions

#### Motivation for adding to Mudlet
fix  #5142
#### Other info (issues closed, discussion etc)

#### Release post highlight

Edru2 committed:
use markdown for image in description

vadi2 commented on src/dlgPackageExporter.cpp:
```diff
@@ -381,6 +382,7 @@ bool dlgPackageExporter::eventFilter(QObject* obj, QEvent* evt)
             for (int i = mDescriptionImages.size() - 1; i >= 0; i--) {
                 QString fname = mDescriptionImages.at(i);
                 QFileInfo info(fname);
+                fname.replace(" ", "%20");
```
Try https://doc.qt.io/qt-5/qstring.html#toHtmlEscaped here, which will cover this and other escapings as well.

vadi2 reviewed (approved):
This works well :+1: 

I found a couple separate issues which are unrelated so not a blocker, <https://github.com/Mudlet/Mudlet/issues/5195> and <https://github.com/Mudlet/Mudlet/issues/5196>

vadi2 committed:
Implement proper escaping

Edru2 commented on src/dlgPackageExporter.cpp:
```diff
@@ -382,7 +382,7 @@ bool dlgPackageExporter::eventFilter(QObject* obj, QEvent* evt)
                 QString fname = mDescriptionImages.at(i);
                 QFileInfo info(fname);
-                fname.replace(" ", "%20");
+                fname = fname.toHtmlEscaped();
```
toHtmlEscaped() unfortunately doesn't seem to convert " " (space) to "%20" which means filenames with a space in it won't work any more with this change

vadi2 replied:
Ah ok.

vadi2 committed:
Try convenience method for escaping

vadi2 committed:
Add debug info

vadi2 committed:
Fix qDebug() syntax

vadi2 commented on src/dlgPackageExporter.cpp:
```diff
@@ -381,6 +382,8 @@ bool dlgPackageExporter::eventFilter(QObject* obj, QEvent* evt)
             for (int i = mDescriptionImages.size() - 1; i >= 0; i--) {
                 QString fname = mDescriptionImages.at(i);
                 QFileInfo info(fname);
+                fname = QUrl::toPercentEncoding(fname).constData();
```
This works well - results in `fname is: "%2FUsers%2Fadministrator%2FDownloads%2Fpicture%20with%20spaces%20in%20name.jpg"
` @Edru2

vadi2 committed:
Remove debugging echoes

Edru2 committed:
convert spaces to %20 also when exporting

forgot to convert spaces to %20 in config.lua which meant that description images with space in it's name werent shown in the package manager

Edru2 committed:
Merge branch 'development' into markdown-for-images-in-packageexporter
